lyricsgenius/api.py

```diff
@@ -276,6 +427,105 @@ def search_artist(self, artist_name, verbose=True, max_songs=None, take_first_re
+            return None
+        # get the html section indicating if the song is missing lyrics
+        missing = soup.find_all('div', attrs={'class': 'chart_row-metadata_element chart_row-metadata_element--large'})
+        miss_nb = 0
+        # count the number of songs without lyrics
+        for miss in missing:
+            if miss.text.find("(Missing Lyrics)") >= 0 or miss.text.find("(Unreleased)") >= 0:
+                miss_nb += 1
+        divi = soup.find_all('div', attrs={'class': 'column_layout-column_span column_layout-column_span--primary'})
+        for div in divi:
+            var = 0
+            # get the html section indicating the track numbers (this will be to eliminate sections similar to those of songs but are actually of tracklist or credits of the album)
+            mdiv = div.find_all('span', attrs={'class': 'chart_row-number_container-number chart_row-number_container-number--gray'})
+            for mindiv in mdiv:
+                nb = mindiv.text.replace("\n", "")
+                if nb != "":
+                    index.append(nb)
+            # create the pandas dataframe holding the tracks' titles
+            df = pd.DataFrame(index=index, columns=['track_title'])
+            ndiv = div.find_all('h3', attrs={'class': 'chart_row-content-title'})
+            for mindiv in ndiv:
+                tt = mindiv.text.replace("\n", "").strip()
+                # getting rid of the featurings in the title
+                if tt.find("(Ft") >= 0:
+                    tt = tt.split(" (Ft.", 1)[0]
+                else:
+                    # getting ride of "lyrics" at the end of the title
+                    tt = tt.rsplit(" ", 1)[0].strip()
+                df['track_title'][var] = tt
+                var += 1
+                if var == len(df.index):
+                    break
+        # loop to add song with title from the dataframe
+        for track in df['track_title']:
+            # search the song
+            song = api.search_song(track, artist.name)
+            # if the song was found, it's added to artist
+            if song != None:
+                artist.add_song(song)
+            # if the song wasn't found, it might be because it's formatted in this way : "title by other artist"
+            elif track.find("by") >= 0:
+                s_artist_name = track.replace("\xa0", " ").rsplit(" by ", 1)[1]
+                s_artist = api.search_artist(s_artist_name, max_songs=0)
+                track = track.replace("\xa0", " ").rsplit(" by ", 1)[0].strip()
+                # look for song with other artist
+                song = api.search_song(track, s_artist.name)
+                if song != None:
+                    # add song to the album's main artist
+                    artist.add_song(song)
+                else:
+                    print("Missing lyrics")
+            else:
+                print("Missing lyrics")
+        artist.save_lyrics(format='json')
```

Comment:
The function shouldn't save lyrics, it should return the `Artist` object (or an `Album` object if we wrote that class).